tests/components/spokestack/test_tts.py

```diff
@@ -0,0 +1,166 @@
+def mock_client():
+    """Mock TextToSpeechClient."""
+    with patch(
+        "homeassistant.components.spokestack.tts.TextToSpeechClient"
+    ) as mock_client:
+        yield mock_client
+
+
+async def test_service_say_default(hass, mock_client, calls):
+    """Test service call say."""
+    await async_setup_component(
+        hass,
+        tts.DOMAIN,
+        {
+            tts.DOMAIN: {
+                "platform": "spokestack",
+                CONF_IDENTITY: "test_key",
+                CONF_SECRET_KEY: "test_secret",
+            }
+        },
+    )
+    await hass.services.async_call(
+        tts.DOMAIN,
+        "spokestack_say",
+        {
+            "entity_id": "media_player.something",
+            tts.ATTR_MESSAGE: "There is a person at the front door.",
+        },
+        blocking=True,
+    )
+    assert len(calls) == 1
+    assert len(mock_client.mock_calls) == 3
+    assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".mp3") != -1
+
+    # Confirm validation call is made.
+    assert mock_client.mock_calls[0][2] == {
+        "key_id": "test_key",
+        "key_secret": "test_secret",
+    }
+    # Confirm synthesize is called with valid arguments.
+    assert mock_client.mock_calls[1][2] == {
+        "utterance": "There is a person at the front door.",
+        "voice": DEFAULT_VOICE,
+        "mode": DEFAULT_MODE,
+        "profile": DEFAULT_PROFILE,
+    }
+
+
+async def test_service_say_ssml(hass, mock_client, calls):
+    """Test service call say."""
+    await async_setup_component(
+        hass,
+        tts.DOMAIN,
+        {
+            tts.DOMAIN: {
+                "platform": "spokestack",
+                CONF_IDENTITY: "test_key",
+                CONF_SECRET_KEY: "test_secret",
+            }
+        },
+    )
+    await hass.services.async_call(
+        tts.DOMAIN,
+        "spokestack_say",
+        {
+            "entity_id": "media_player.something",
+            tts.ATTR_MESSAGE: "There is a person at the front door.",
+            tts.ATTR_OPTIONS: {"mode": "ssml"},
+        },
+        blocking=True,
+    )
+    assert len(calls) == 1
+    assert len(mock_client.mock_calls) == 3
+    assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".mp3") != -1
+
+    # Confirm validation call is made.
+    assert mock_client.mock_calls[0][2] == {
+        "key_id": "test_key",
+        "key_secret": "test_secret",
+    }
+    # Confirm synthesize is called with valid arguments.
+    assert mock_client.mock_calls[1][2] == {
+        "utterance": "There is a person at the front door.",
+        "voice": DEFAULT_VOICE,
+        "mode": "ssml",
+        "profile": DEFAULT_PROFILE,
+    }
+
+
+async def test_service_say_error(hass, mock_client, calls):
+    """Test service call say with http response 400."""
+    mock_client.synthesize.return_value = TTSError
+
+    await async_setup_component(
+        hass,
+        tts.DOMAIN,
+        {
+            tts.DOMAIN: {
+                "platform": "spokestack",
```

Comment:
This test is all wrong. The spokestack platform is set up via a config entry and not with platform config.

Reply:
Since the spokestack platform is set up via a config entry does that mean that our tests should be different than all of the other tts platforms? Where it calls `async_setup_entry` instead of `async_setup_component`?

Reply:
Correct.